activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -407,6 +407,13 @@ def initialize(pool_config)
 
         @lock_thread = false
 
+        @background_executor = Concurrent::ThreadPoolExecutor.new(
+          min_threads: 0,
+          max_threads: @size,
```

Comment:
Somewhat related to @fxn comment, and probably better addressed in a followup, but I think this is a bad default.

I think we should make sure the background executor can never starve foreground threads. Maybe the background executor should have it's own "pool_size" config to both set the number of threads and increase the connection pool, e.g.:

```yaml
production:
  adapter: ...
  pool: <%= ENV['RAILS_MAX_THREADS'] %>
  async_pool: 12
```